### PR TITLE
midv1.4 additions

### PR DESCRIPTION
**Pull request by RusdiHaizim**
+Reformatted deadline format, CONFIRMED! 
+refactored help commands

### CODE REVIEW

File: src/main/java/javacake/commands/EditCommand.java [hunk 1]
```diff
@@ -33,11 +33,17 @@ public String execute(Logic logic, Ui ui, StorageManager storageManager) throws
             System.out.println("Task: " + input);
             String[] buffer = input.split("\\s+");
             System.out.println("Task num: " + buffer[1]);
-            int num = Integer.parseInt(buffer[1]);
-            --num;
+            int taskIdx = Integer.parseInt(buffer[1]);
```
Comment (GlenWong97) [inline src/main/java/javacake/commands/EditCommand.java]: variable taskIdx can be refactored into taskIndex to improve readability and ambiguity.

File: src/main/java/javacake/tasks/Deadline.java [hunk 2]
```diff
@@ -37,6 +36,7 @@ public Deadline(String description, String by) throws CakeException {
             SimpleDateFormat simpleDateFormat = new SimpleDateFormat(formatDate);
             simpleDateFormat.setLenient(false);
             dateNow = simpleDateFormat.parse(by);
+            this.by = by;
```
Comment (GlenWong97) [inline src/main/java/javacake/tasks/Deadline.java]: `by` can be renamed as mentioned by TA. Maybe it can be `deadLineDate`.

**Review by GlenWong97 — APPROVED** [review]
Great job trying to fix the natty library problem. Do take note on some minor feedback.

File: src/main/java/javacake/tasks/Deadline.java (hunk 2 above)
Comment (RusdiHaizim) [reply src/main/java/javacake/tasks/Deadline.java]: rectified with better name.

**Pull request merged**